lib/entice/logic/MapInstance.ex

```diff
@@ -0,0 +1,79 @@
+defmodule Entice.Logic.MapInstance do
+  alias Entice.Entity
+  alias Entice.Logic.MapInstance
+  alias Entice.Logic.Npc
+
+
+  defstruct(
+    players: [],
+    npcs: [], #TODO: Figure out if we want the npcs ot be able to join&leave or stay there even if dead or unspawned
+    map: nil)
+
+  @doc """
+  Registers a MapInstance.Behaviour for the Entity.
+  Params:
```

Comment:
we can go with elixir source code doc conventions here... i'll see if i can find a good example in the elixir docs and will come back to you with it :)